src/builder/BridgeRoutes.sol

```diff
@@ -0,0 +1,465 @@
+        uint256[] chainIds;
+        uint256[] maxCosts;
+    }
+
+    /* ===== Output Types ===== */
+
+    struct BuilderResult {
+        // version of the builder interface. (Same as VERSION, but attached to the output.)
+        string version;
+        // array of quark operations to execute to fulfill the client intent
+        QuarkWallet.QuarkOperation[] quarkOperations;
+        // array of action context and other metadata corresponding 1:1 with quarkOperations
+        QuarkAction[] quarkActions;
+        // EIP-712 digest to sign for a MultiQuarkOperation to fulfill the client intent.
+        // Empty when quarkOperations.length == 0.
+        bytes multiQuarkOperationDigest;
+        // EIP-712 digest to sign for a single QuarkOperation to fulfill the client intent.
+        // Empty when quarkOperations.length != 1.
+        bytes quarkOperationDigest;
+        // client-provided paymentCurrency string that was used to derive token addresses.
+        // client may re-use this string to construct a request that simulates the transaction.
+        string paymentCurrency;
+    }
+
+    // With QuarkAction, we try to define fields that are as 1:1 as possible with the
+    // simulate endpoint request schema.
+    struct QuarkAction {
+        uint256 chainId;
+        string actionType;
+        bytes actionContext;
+        // One of the PAYMENT_METHOD_* constants.
+        string paymentMethod;
+        // Address of payment token on chainId.
+        // Null address if the payment method was OFFCHAIN.
+        address paymentToken;
+        uint256 paymentMaxCost;
+    }
+
+    struct TransferActionContext {
+        uint256 amount;
+        uint256 price;
+        address token;
+        uint256 chainId;
+        address recipient;
+    }
+
+    struct BridgeActionContext {
+        uint256 amount;
+        uint256 price;
+        address token;
+        uint256 chainId;
+        address recipient;
+        uint256 destinationChainId;
+    }
+
+
+    /* ===== Internal/Intermediate Types ===== */
+
+    struct AssetPositionsWithChainId {
+        uint256 chainId;
+        address asset;
+        string symbol;
+        uint256 decimals;
+        uint256 usdPrice;
+        AccountBalance[] accountBalances;
+    }
+
+    enum BridgeType {
+        NONE,
+        CCTP
+    }
+
+    struct Bridge {
+        // Note: Cannot name these `address` nor `type` because those are both reserved keywords
+        address bridgeAddress;
+        BridgeType bridgeType;
+    }
+
+    // TODO: convert strings to lower case before comparing. maybe rename to `compareSymbols`
+    function compareStrings(string memory a, string memory b) internal pure returns (bool) {
+        return keccak256(abi.encodePacked((a))) == keccak256(abi.encodePacked((b)));
+    }
+
+    function filterAssetPositionsForSymbol(string memory assetSymbol, ChainAccount[] memory chainAccountsList) internal pure returns (AssetPositionsWithChainId[] memory) {
```

Comment:
Nb. `ForSymbol` seems unnecessary here. Supposing we would support some other way of filtering, I'm not sure what it would be -- positions are already organized by chain ID in the input.

Reply:
Could maybe filter based on `account` if we want to leave certain subaccounts untouched. But fair enough, can shorten the name for now.

Reply:
Done